dedupe/labeler.py

```diff
@@ -69,24 +69,22 @@ def candidates(self) -> TrainingExamples:  # type: ignore[override]
     def candidates(self, new_candidates: TrainingExamples) -> None:
         self._candidates = new_candidates
 
-        self.distances = self.transform(self._candidates)
+        self.distances = self._distances(self._candidates)
 
         random_pair = random.choice(self._candidates)
         exact_match = (random_pair[0], random_pair[0])
-        self.fit_transform([exact_match, random_pair], [1, 0])
+        self.fit([exact_match, random_pair], [1, 0])
 
-    def transform(self, pairs: TrainingExamples) -> numpy.typing.NDArray[numpy.float_]:
+    def _distances(self, pairs: TrainingExamples) -> numpy.typing.NDArray[numpy.float_]:
         return self.data_model.distances(pairs)
 
-    def fit(self, X: numpy.typing.NDArray[numpy.float_], y: LabelsLike) -> None:
-
+    def _fit(self, X: numpy.typing.NDArray[numpy.float_], y: LabelsLike) -> None:
         self.y: numpy.typing.NDArray[numpy.int_] = numpy.array(y)
         self.X = X
+        sklearn.linear_model.LogisticRegression.fit(self, self.X, self.y)
```

Comment:
i think it's good to have an explicit call here, but if we do that, we shouldn't also have the class inherit from `sklearn.linear_model.LogisticRegression`

Reply:
If I turn this to `super().fit(self.X, self.y)` then mypy complains because it needs both parents in the inheritance to have the same signature for `fit`: `dedupe/labeler.py:84: error: Argument 1 to "fit" of "Learner" has incompatible type "ndarray[Any, dtype[floating[_64Bit]]]"; expected "List[Tuple[Mapping[str, Any], Mapping[str, Any]]]"  [arg-type]`

I think we should rename `RLRLearner` to `MatchLearner` (it is comparing records using distances, and this is sorta a nice opposition to BlockLEarner) and make it wrap `LogisticRegression` (or RandomForest, or whatever) using composition, not inheritance.

Reply:
See if you like what I did.